libsolidity/parsing/Parser.cpp

```diff
@@ -333,8 +333,16 @@ Parser::FunctionHeaderParserResult Parser::parseFunctionHeader(bool _forceEmptyN
 {
 	RecursionGuard recursionGuard(*this);
 	FunctionHeaderParserResult result;
-	expectToken(Token::Function);
-	if (_forceEmptyName || m_scanner->currentToken() == Token::LParen)
+
+	if (m_scanner->currentToken() == Token::Function)
+		result.isConstructor = false;
+	else if (m_scanner->currentToken() == Token::Constructor)
```

Comment:
Unfortunately the PR this way cannot be merged before 0.5.0, but as a workaround we could check for current literal and if that is `constructor` accept it.

Later in 0.5.0 we can introduce the keyword and update the parser. Check out how the event `emit` was implemented, it is the same idea.